### PR TITLE
fix isssues detected by coverity

### PR DESCRIPTION
**Pull request by lum1n0us**

### CODE REVIEW

File: samples/socket-api/wasm-src/udp_client.c [hunk 1]
```diff
@@ -73,6 +73,7 @@ main(int argc, char *argv[])
                    (struct sockaddr *)&server_address, &serverlen);
 
     if (ret > 0) {
+        buffer[ret] = '\0';
```
Comment (wenyongh) [inline samples/socket-api/wasm-src/udp_client.c:76]: Will ret be equal to sizeof(buffer)? If yes, buffer[ret] will cause illegal memory access.
How about change to:
```
char msg[64];
snprintf(msg, sizeof(msg), "[Client] Buffer recieved: %%.%ds\n", ret);   => construct string "%.ns", n == ret
printf(msg, buffer);
```

File: samples/socket-api/wasm-src/udp_server.c [hunk 2]
```diff
@@ -79,6 +79,7 @@ main(int argc, char *argv[])
             perror("Read failed");
             goto fail;
         }
+        buffer[ret] = '\0';
```
Comment (wenyongh) [inline samples/socket-api/wasm-src/udp_server.c:82]: Same as above

File: samples/socket-api/wasm-src/udp_client.c (hunk 1 above)
Comment (lum1n0us) [reply samples/socket-api/wasm-src/udp_client.c:76]: according to code `core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c`,  it * returns the number of bytes read or -1 for errors.*.

**Comment by loganek** [comment]
@lum1n0us @wenyongh is it possible to run coverity as part of the PR check?

**Comment by wenyongh** [comment]
> @lum1n0us @wenyongh is it possible to run coverity as part of the PR check?

Maybe, we will check it in the future. To be frankly, it takes long time to compile all the possible feature combinations and not sure whether it is suitable to make it as a CI.

**Pull request merged**